sympy/sets/tests/test_fancysets.py

```diff
@@ -295,6 +296,114 @@ def test_Range_set():
         assert r.inf == rev.inf and r.sup == rev.sup
         assert r.step == -rev.step
 
+    # test symbolic Range
+    n = symbols('n', integer=True)
+    r = Range(1, n)
+    assert r.subs(n, -2) == Range(0, 0, 1)
+    assert r.subs(n, 3) == Range(1, 3, 1)
+    ranges = [ Range(n, 10, 1), Range(n, -10, 1), Range(n, -10, -1),
+               Range(n, 10, -1), Range(1, n, 1), Range(-1, n, 1),
+               Range(1, n, -1), Range(-1, n, -1), Range(1, 5, n),
+               Range(-1, 5, n), Range(1, -5, n), Range(-1, -5, n) ]
+    expected = [(Range(9, n - 1, -1), [Range(9, -2, -1), Range(9, -1, -1), Range(9, 0, -1)]),
+                (Range(-11, n - 1, -1), [Range(0, 0, 1), Range(0, 0, 1), Range(0, 0, 1)]),
+                (Range(-9, n + 1, 1), [Range(-9, 0, 1), Range(-9, 1, 1), Range(-9, 2, 1)]),
+                (Range(11, n + 1, 1), [Range(0, 0, 1), Range(0, 0, 1), Range(0, 0, 1)]),
+                (Range(n - 1, 0, -1), [Range(0, 0, 1), Range(0, 0, 1), Range(0, 0, 1)]),
+                (Range(n - 1, -2, -1), [Range(0, 0, 1), Range(-1, -2, -1), Range(0, -2, -1)]),
+                (Range(n + 1, 2, 1), [Range(0, 2, 1), Range(1, 2, 1), Range(0, 0, 1)]),
+                (Range(n + 1, 0, 1), [Range(0, 0, 1), Range(0, 0, 1), Range(0, 0, 1)]),
+                (Range(5 - n, -n*ceiling(4/n) - n + 5, -n), [Range(0, 0, 1), Range(4, 0, -1)]),
+                (Range(5 - n, -n*ceiling(6/n) - n + 5, -n), [Range(0, 0, 1), Range(4, -2, -1)]),
+                (Range(-n - 5, -n*ceiling(-6/n) - n - 5, -n), [Range(-4, 2, 1), Range(0, 0, 1)]),
+                (Range(-n - 5, -n*ceiling(-4/n) - n - 5, -n), [Range(-4, 0, 1), Range(0, 0, 1)])]
+    for r, expec in zip(ranges, expected):
+        rev = r.reversed
+        if not r.step.has(Symbol):
+            assert (rev, [rev.subs(n, -1), rev.subs(n, 0), rev.subs(n, 1)]) == expec
+        else:
+            assert (rev, [rev.subs(n, -1), rev.subs(n, 1)]) == expec
+    expected = [(Piecewise((True, n <= 2), (False, True)), True, True, True),
```

Comment:
This is going to be a fragile test. What about:
```python
for r in ranges:
    rc = r._contains(2)
    t = Tuple(*r.args)
    for v in (-1, 0, 1):
        assert rc.subs(n, v) == (2 in list(range(*t.subs(n,v))))
```

Reply:
Ah! Your tests are more elegant, I will rewrite the tests.

Reply:
necessity is the mother of invention :-)

Reply:
I am re-writing the tests.